### DISCUSSION TITLE
How to achieve this heading behaviour?

### DISCUSSION THREAD

**Discussion by th0mr** [brechtm/rinohtype#399] [Q&A] [answered]
I am currently trying to achieve this styling for my headings, i.e. the label for the header is sent back 2cm from the rest of the content but all lines of the heading content remain where they are. 

Here is an edited image showing what I would like the styling to look like.
![expected_output](https://user-images.githubusercontent.com/87852659/216413662-f58b4dbb-fbae-4e89-977e-c95a2047c8cc.jpg)

Below is my current output using my stylesheet. As you can see, when the heading becomes multiple lines long then the heading does not remain in the "content" section of the document like I illustrated in the picture above.

![problematic paragraphs](https://user-images.githubusercontent.com/87852659/216416478-e7ddf6a0-41f9-425b-9c4f-d987db57ab0d.png)

I'm doing this with a custom stylesheet. I have created a minimal example that replicates this issue:

```
[STYLESHEET]
name=min_example
description=Minimal example of long title issue
pygments_style=friendly
base=sphinx

[chapter]
# All content needs to be pushed 2cm right
margin_left=2cm

[heading level 1]
# These options below create the heading as 1.2.3 -TAB- title
# But at the moment the problem exists that only the first line is tabbed in, making 2+ line titles look odd
#
# Pushes the title back 2cm to move left past the chapter content
margin_left=-2cm
# adds tab stop between heading number and heading
tab_stops=2cm LEFT
# adds tab character between tab stop and heading to create the gap
before="\t"
```

Do you have any idea how I might be able to solve this? In an ideal world I would apply a style to the heading label to send it back 2cm, but as far as I can read in the rinohtype docs, this does not seem possible.

I had a quick look at the stylesheets for the rinohtype docs, where you are trying to achieve something similar for the styling of your headings. https://github.com/brechtm/rinohtype/blob/master/doc/rinohtype.rts. But as far as I can tell, the same problem would occur if your titles went past two lines.

Any advice would be greatly appreciated 😄

**Comment by brechtm** [comment, marked as answer]
> Do you have any idea how I might be able to solve this? In an ideal world I would apply a style to the heading label to send it back 2cm, but as far as I can read in the rinohtype docs, this does not seem possible.

In fact, that is exactly how you can solve this, and yes, it's possible :-)

```ini
[heading level 1]
#margin_left=0cm
# adds tab character between tab stop and heading to create the gap
before="\t"
# set tab stop at the margin
tab_stops=0cm LEFT
# create room for the section label
indent_first=-2cm
```

**Reply by th0mr** [reply]
I've just rebuilt the document with this styling and that fixes my issue.

Thank you for the help! 👍
